rdrf/rdrf/helpers/utils.py

```diff
@@ -302,6 +302,18 @@ def consent_status_for_patient(registry_code, patient):
     return all(sections[section_code].is_valid(section_answers) for section_code, section_answers in answers.items())
 
 
+def consent_status_for_patient_consent(registry, patient_id, consent_question_code):
+    from rdrf.models.definition.models import ConsentQuestion
+    from registry.patients.models import ConsentValue
+
+    consent_questions = ConsentQuestion.objects.filter(section__registry=registry,
+                                                       code=consent_question_code)
+    consents_accepted_cnt = [ConsentValue.objects.filter(consent_question__id=consent_question.id, patient_id=patient_id, answer=True).count()
+                             for consent_question in consent_questions]
+
+    return all(consents_accepted_cnt)
```

Comment:
Just a note that `all([]) == True` (ie. when it can't find that consent code), which is fine for this usecase, but in a util method might be unexpected behaviour.

Reply:
Fixed and added a unit test to catch it if we remove the logic for some reason.